fix: remove modal when navigating back

mohit-nagaraj opened the pull request:
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #5008 

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

https://github.com/user-attachments/assets/f6dc99ac-5727-4143-90b8-bf105ccf3a7e

mohit-nagaraj commented:
Do let me kno if other pop ups also needs these life cycle events to be implemented

CommanderStorm commented on src/components/CreateGroupDialog.vue:
```diff
@@ -33,15 +33,27 @@
         modal: null,
         groupName: null,
     }),
+    watch: {
+        $route(to, from) {
+            this.cleanupModal();
+        }
+    },
```
is this not redundant to `beforeRouteLeave`?

```suggestion
```

mohit-nagaraj replied:
hmm yea i was trying out both ways seeing which one worked then just left both off them there. beforeRouteLeave should handle the modal cleanup during route transitions then will remove any other redundant code

CommanderStorm commented on src/components/CreateGroupDialog.vue:
```diff
@@ -33,15 +33,27 @@
 import { Modal } from "bootstrap";
 
 export default {
+    beforeRouteLeave(to, from, next) {
```
are there cases where the `beforeUnmount` is not called?
If yes, please add a comment what this case is.
If no, lets remove this check and inline `cleanupModal` into `beforeUnmount`

mohit-nagaraj replied:
If the user refreshes the page or navigates to another URL using the browser's address bar, the Vue instance might not clean up properly, meaning beforeUnmount won't be triggered. correct me if i am wrong

mohit-nagaraj replied:
btw all of cleanupModal shud be made inline right if i am not wrong

mohit-nagaraj replied:
hey so only one hook is calling the function cleanupModal, shud i make it inline? @CommanderStorm

CommanderStorm commented on src/components/CreateGroupDialog.vue:
```diff
@@ -58,6 +70,22 @@ export default {
+                    console.warn("Modal hide failed:", e);
+                }
+            }
+            document.body.classList.remove("modal-open");
+            document.body.style.paddingRight = "";
+            document.body.style.overflow = "";
```
as far as I see these are already removed by `this.modal.hide()`.
Is there a case where you needed them which I am missing?

```suggestion
```

mohit-nagaraj replied:
nope it wasnt actually removing those details, so thought to manually remove them

mohit-nagaraj replied:
nvm now tht i removed them still its working properly now too :)

CommanderStorm reviewed (changes requested):
Thanks for the PR,

I have left a few questions below, mostly regarding a bit of likely code-duplication.

mohit-nagaraj commented:
> Feel too hacky and unsafe, as it's utilized dom and added a lot of events. I prefer not to change the current logic.

hey i have updated the code now, does this make it better.... i am just now relying on beforeUnmount hook to remove the modal

mohit-nagaraj commented:
@louislam could you re-review this once

mohit-nagaraj commented:
@CommanderStorm

mohit-nagaraj commented:
Hey @louislam take a look into this pr once free

CommanderStorm commented on .deepsource.toml:
```diff
@@ -0,0 +1,31 @@
+version = 1
```
I don't see why we should add whatever this file is
Please remove this file

mohit-nagaraj replied:
sure sure it was my working changes tht got committed here sorry

CommanderStorm commented on extra/healthcheck.go:
```diff
@@ -32,20 +32,20 @@ func main() {
 
 	sslKey := os.Getenv("UPTIME_KUMA_SSL_KEY")
-	if len(sslKey) == 0 {
+	if sslKey == "" {
```
Entirely unrelated change.
Please revert this file

mohit-nagaraj replied:
same thing with this as well will remove it

CommanderStorm reviewed (changes requested):
Please remove the unrelated changes.
Otherwise this looks good to me

mohit-nagaraj commented:
Done removed those changes

mohit-nagaraj commented:
Hey @CommanderStorm can u review this when u get time 
Thank again

mohit-nagaraj commented:
@CommanderStorm  could you review it when u get time

mohit-nagaraj commented:
@CommanderStorm when is the earliest possible time u can retake a look in this again

CommanderStorm reviewed (approved):
Thanks for the work and the bugfix ❤️ 
Sorry for the long wait
